optimum/onnxruntime/base.py

```diff
@@ -277,8 +281,11 @@ def compute_past_key_values_output_shapes(
             `Dict[str, List[int]]`: The dictionary mapping each past key value output name to its corresponding shape.
         """
         batch_size = input_ids.size(0)
-        num_attention_heads = self.normalized_config.num_attention_heads
-        embed_size_per_head = self.normalized_config.hidden_size // num_attention_heads
+        if self.normalized_config.config.model_type in {"mistral", "llama"}:
+            num_attention_heads = self.normalized_config.num_attention_heads
+        else:
+            num_attention_heads = self.normalized_config.num_key_value_heads
+        embed_size_per_head = self.normalized_config.hidden_size // self.normalized_config.num_attention_heads
```

Comment:
rather:

```suggestion
        num_attention_heads = self.normalized_config.num_attention_heads
        if self.normalized_config.config.model_type in {"mistral", "llama"}:
            embed_size_per_head = self.normalized_config.hidden_size // self.normalized_config.num_key_value_heads
        else:
            embed_size_per_head = self.normalized_config.hidden_size // num_attention_heads
```

is that not? Referring to https://github.com/huggingface/transformers/blob/87499420bff74122a404d1bb0db2e57e6d1ecfe9/src/transformers/models/llama/modeling_llama.py#L352-L353

Reply:
here `self.head_dim = config.hidden_size // config.num_attention_heads`
https://github.com/huggingface/transformers/blob/87499420bff74122a404d1bb0db2e57e6d1ecfe9/src/transformers/models/llama/modeling_llama.py#L270

Reply:
Yes my bad!